### PR TITLE
Refactor send row result

### PR DESCRIPTION
**Pull request by CJKkkk-315**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #12390 

## What this PR does / why we need it:

清理了返回查询结构相关代码的无用逻辑和参数传递
修改了部分函数名，提高可读性
binary传输中，将写入NullBitMap的逻辑单独独立成函数
修正了TIME类数据在二进制传输中，负值无法正确显示的问题

___

### **PR Type**
Bug fix, Enhancement

___

### **Description**
- Removed unused parameters from functions like `writePackets`, `appendIntLenEnc`, `appendCountOfBytes`, `appendStringFix`, `appendStringLenEnc`, `appendCountOfBytesLenEnc`, `appendStringLenEncOfInt64`, `appendStringLenEncOfUint64`, `appendStringLenEncOfFloat64`, `appendUint8`, `appendUint16`, `appendUint32`, `appendUint64`, `openRow`, `closeRow`, `flushOutBuffer`, `openPacket`, `closePacket`, and `append`.
- Refactored functions to improve readability and maintainability, including renaming and restructuring.
- Fixed the issue with negative TIME values in binary transmission by correctly parsing and appending time values.
- Consolidated logic for appending data to buffers, reducing redundancy and potential errors.
- Removed debug count increments which were not necessary.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>mysql_protocol.go</strong><dd><code>Refactor and bug fixes in MySQL protocol implementation.</code>&nbsp; </dd></summary>
<hr>
      
pkg/frontend/mysql_protocol.go

<li>Removed unused parameters from several functions.<br> <li> Refactored functions to improve readability and maintainability.<br> <li> Fixed the issue with negative TIME values in binary transmission.<br> <li> Consolidated logic for appending data to buffers.<br>

</details>
    

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/16780/files#diff-df9bbc34115212918609e47ad31cf72bf7b2be07d21c34c045074def608d0462">+210/-241</a></td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: pkg/frontend/mysql_protocol.go [hunk 1]
```diff
@@ -2531,54 +2531,42 @@ func (mp *MysqlProtocolImpl) WriteResultSetRow(mrs *MysqlResultSet, cnt uint64)
...
 		mp.flushCount++
 		mp.writeBytes += uint64(mp.bytesInOutBuffer)
 		// FIXME: use a suitable timeout value
-		mp.incDebugCount(8)
 		mp.ses.CountPacket(1)
 		err := mp.tcpConn.Flush(0)
-		mp.incDebugCount(9)
 		if err != nil {
-			return err
+			return
```
Comment (daviszhen) [inline pkg/frontend/mysql_protocol.go:2561]: 这个err 不能吞掉吧。

File: pkg/frontend/mysql_protocol.go [hunk 2]
```diff
@@ -2587,40 +2575,45 @@ func (mp *MysqlProtocolImpl) openPacket() error {
...
 	writeIdx += n
 	mp.bytesInOutBuffer += n
 	outbuf.SetWriteIndex(writeIdx)
-	return nil
+	return
 }
 
-func beginWriteIndex(outbuf *goetty_buf.ByteBuf, offset int) int {
-	if offset < 0 {
+//func beginWriteIndex(outbuf *goetty_buf.ByteBuf, offset int) int {
```
Comment (daviszhen) [inline pkg/frontend/mysql_protocol.go:2586]: 删除

File: pkg/frontend/mysql_protocol.go [hunk 3]
```diff
@@ -2635,130 +2628,116 @@ func (mp *MysqlProtocolImpl) fillPacket(elems ...byte) error {
...
-		return nil
+		return
 	}
 	outbuf := mp.tcpConn.OutBuf()
-	payLoadLen := outbuf.GetWriteIndex() - beginWriteIndex(outbuf, mp.beginOffset) - 4
+	payLoadLen := outbuf.GetWriteIndex() - mp.getBeginWriteIndex() - 4
 	if payLoadLen < 0 || payLoadLen > int(MaxPayloadSize) {
-		return moerr.NewInternalError(mp.ctx, "invalid payload len :%d curWriteIdx %d beginWriteIdx %d ",
-			payLoadLen, outbuf.GetWriteIndex(), beginWriteIndex(outbuf, mp.beginOffset))
+		return
```
Comment (daviszhen) [inline pkg/frontend/mysql_protocol.go:2650]: err 不能吞

File: pkg/frontend/mysql_protocol.go [hunk 4]
```diff
@@ -2635,130 +2628,116 @@ func (mp *MysqlProtocolImpl) fillPacket(elems ...byte) error {
...
-		data = mp.append(data, value.Month(), value.Day())
+		mp.append(4)
+		mp.appendUint16(value.Year())
+		mp.append(value.Month(), value.Day())
 	}
-	return data
 }
 
-// the server send every row of the result set as an independent packet
+// the server send every row of the result set as an independent packetjjiji
```
Comment (daviszhen) [inline pkg/frontend/mysql_protocol.go:2740]: ？？？

File: pkg/frontend/mysql_protocol.go [hunk 5]
```diff
@@ -2531,54 +2531,42 @@ func (mp *MysqlProtocolImpl) WriteResultSetRow(mrs *MysqlResultSet, cnt uint64)
...
-		return err
-	}
-
-	err = mp.flushOutBuffer()
-	if err != nil {
-		return err
-	}
-	return err
+func (mp *MysqlProtocolImpl) closeRow() {
+	mp.closePacket(true)
```
Comment (daviszhen) [inline pkg/frontend/mysql_protocol.go:2547]: 检查下error

**Review by daviszhen — CHANGES_REQUESTED** [review]

**Comment by CJKkkk-315** [comment]
还原了部分不可删除的err。 修改了test文件中的一些对应代码

**Review by daviszhen — APPROVED** [review]

**Pull request merged**